src/quad_gl.rs

```diff
@@ -738,7 +781,7 @@ impl QuadGl {
             .iter_mut()
             .zip(self.draw_calls_bindings.iter_mut())
         {
-            let pipeline = self.pipelines.get_quad_pipeline(dc.pipeline);
+            let pipeline = self.pipelines.get_quad_pipeline_mut(dc.pipeline);
```

Comment:
I am concerned about this line. Is pipeline must be immutable, and we should clone buffer on lines 829-832?